### ISSUE TITLE
Partial properties on struct warn about field ordering

### ISSUE THREAD

**Issue by sbomer** [dotnet/roslyn#74679]
[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAFxFANgHwAICYCMBYAKAAcIAnBKCFAAgGcFSBXAYwWoGVqBvQ6v6kuUo1YbAJLdqAcwCmCANzUAvoRUFCgilToMWbTjwL9qGAMwcAFAEpqAXgB81KHeoAWLPN78ipKADcIBBlqUhkIABMAexgUAE8nGDYoTyNvMi0RROoJBycUtSA==)

```csharp
partial struct S {
    partial int I { get; }
}

partial struct S {
    public S() => i = 42;
    private readonly int i;
    partial int I => i;
}
```

Expected behavior: no warning. I should be able to implement a partial property on a struct with my own backing field.

Actual behavior:
```
warning CS0282: There is no defined ordering between fields in multiple declarations of partial struct 'S'. To specify an ordering, all instance fields must be in the same declaration.
```
